<?xml version="1.0" encoding="utf-8"?>

<launch>
  <!-- Console launch prefix -->
  <arg name="image" default="/usb_cam/image_raw" />

  <!-- ROS parameter files -->
  <arg name="ros_param_file"             default="$(find scale_truck_control)/config/config.yaml"/>
  <arg name="lrc_param_file"             default="$(find scale_truck_control)/config/lrc.yaml"/>
  <arg name="vehicle_param_file"             default="$(find scale_truck_control)/config/LV.yaml"/>
  <arg name="lidar_param_file"             default="$(find scale_truck_control)/config/laser_filter.yaml"/>

  <!-- Load parameters -->
  <rosparam command="load" ns="scale_truck_control" file="$(arg ros_param_file)"/>
  <rosparam command="load" ns="scale_truck_control" file="$(arg vehicle_param_file)"/>
  <rosparam command="load" ns="LRC" file="$(arg lrc_param_file)"/>
  <rosparam command="load" ns="LRC" file="$(arg vehicle_param_file)"/>


  <!-- Start Scale Truck Control -->
  <node pkg="scale_truck_control" type="scale_truck_control" name="scale_truck_control" output="screen" />
  <node pkg="scale_truck_control" type="LRC" name="LRC" output="screen" /> 
  <node pkg="rosserial_python" type="serial_node.py" name="serial_node" >
    <param name="port" value="/dev/ttyACM0"/>
    <param name="baud" value="57600"/>
  </node>

</launch>
